### PR TITLE
:sparkles: Basic users service

### PR DESCRIPTION
**Pull request by 7rebux**

### CODE REVIEW

**Comment by 7rebux** [comment]
⚠️ Needs to be tested

File: src/auth0.ts [hunk 1]
```diff
@@ -1,46 +1,72 @@
...
-  switch (auth0Response.status) {
-    case 200:
-      return true;
-    case 404:
+  if (response.status === 200) {
+    // TODO: Add necessary data to user object
+    const data: Auth0UserResponse = await response.json();
+    return data;
+  } else {
+    throw new Auth0Error(response.status, await response.text());
```
Comment (HurricanKai) [inline src/auth0.ts:57]: Please don't use exceptions for control flow

File: src/services/users_v1.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  const authContext = await authenticateUser(request.headers);
+
+  switch (request.method) {
+    case 'GET': {
+      if (subpath === 'me') {
+        return await dataById(authContext.userId, env);
+      }
+
+      // TODO:  Is this user data public or are only friends
+      //        allowed to receive it?
```
Comment (HurricanKai) [inline src/services/users_v1.ts:15]: I have a draft for a friends V2 that can handle this. For now it's ok to do it like this.

File: src/services/users_v1.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import { fetchUser } from '../auth0';
+
+export default async (request: Request, env: Env, subpath: string): Promise<Response> => {
+  const authContext = await authenticateUser(request.headers);
+
+  switch (request.method) {
+    case 'GET': {
+      if (subpath === 'me') {
+        return await dataById(authContext.userId, env);
+      }
```
Comment (HurricanKai) [inline src/services/users_v1.ts:12]: Are we sure the JWT doesn't just include this?

File: src/auth0.ts [hunk 4]
```diff
@@ -1,46 +1,72 @@
...
+  family_name: string;
+};
+
+class Auth0Error extends Error {
+  constructor(public code: number, message: string) {
+    super(message);
+  }
+}
+
+const fetchToken = async (env: Env): Promise<Auth0TokenResponse> => {
```
Comment (HurricanKai) [inline src/auth0.ts:23]: TODO: Cache this locally

**Review by HurricanKai — CHANGES_REQUESTED** [review]

File: src/services/users_v1.ts (hunk 3 above)
Comment (7rebux) [reply src/services/users_v1.ts:12]: The JWT doesnt include all data from the auth0 endpoint. Might be deleted once we make sure that no special fields are needed

**Review by HurricanKai — APPROVED** [review]

**Pull request merged**